server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -0,0 +1,530 @@
+                    assertFalse(TestClusterNode.this.node.equals(node));
+                    try {
+                        SetOnce<TransportResponse> responseSetOnce = new SetOnce<>();
+                        TransportChannel channel = mock(TransportChannel.class);
+                        doAnswer(invocation -> {
+                                try {
+                                    // We expect to only send one response per request
+                                    responseSetOnce.set((TransportResponse) invocation.getArguments()[0]);
+                                    handleResponse(requestId, responseSetOnce.get());
+                                    return null;
+                                } catch (Exception e) {
+                                    throw new AssertionError(e);
+                                }
+                            }
+                        ).when(channel).sendResponse(any(TransportResponse.class));
+                        final Runnable transportAction = () -> {
+                            try {
+                                testClusterNodes.nodes.get(node.getName()).mockTransport.getRequestHandler(action)
+                                    .processMessageReceived(request, channel);
+                            } catch (Exception e) {
+                                throw new AssertionError(e);
+                            }
+                        };
+                        // Don't run recovery actions in the deterministic task queue since they can block
+                        if (action.startsWith("internal:index/shard/recovery/")) {
+                            transportAction.run();
+                        } else {
+                            deterministicTaskQueue.scheduleNow(transportAction);
+                        }
+                    } catch (Exception e) {
+                        throw new AssertionError(e);
+                    }
+                }
+            };
+            transportService = mockTransport.createTransportService(
+                settings, deterministicTaskQueue.getThreadPool(
+                    runnable -> () -> {
+                        try (CloseableThreadContext.Instance ignored =
+                                 CloseableThreadContext.put("nodeId", '{' + node.getId() + "}{" + node.getEphemeralId() + '}')) {
+                            runnable.run();
+                        }
+                    }
+                ), NOOP_TRANSPORT_INTERCEPTOR,
+                a -> node, null, emptySet()
+            );
+            final IndexNameExpressionResolver indexNameExpressionResolver = new IndexNameExpressionResolver();
+            repositoriesService = new RepositoriesService(
+                settings, clusterService, transportService,
+                Collections.singletonMap(FsRepository.TYPE, metaData -> {
+                        final Repository repository = new FsRepository(metaData, createEnvironment(), xContentRegistry()) {
```

Comment:
I think the reason this did not fail yet is that we aren't writing any data yet because all the shards are empty?

Regardless, I cleaned this up and made sure all nodes have the same repository path in their settings now :)